### PR TITLE
Fixes RNR screen initial load and first-run UX on login screen

### PR DESCRIPTION
**Pull request by robinheinze**
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant

## Describe your PR

- Adds a default email / password to the auth store which allows the user to just click login when they first boot up their Ignited app. 
- Adds observer to the RNR episodes screen to fix the initial data load bug
- Fixes the ESlint no unused var rule to allow _ prefixing to ignore

### Visuals
Screen Shot 2022-08-22 at 11 29 42 AM

### CODE REVIEW

**Review by frankcalise — APPROVED** [review]

File: boilerplate/app/screens/DemoPodcastListScreen.tsx [hunk 1]
```diff
@@ -27,7 +30,7 @@ export function DemoPodcastListScreen(props: DemoTabScreenProps<"DemoPodcastList
   return (
     <Screen preset="fixed" safeAreaEdges={["top"]}>
       <FlatList<Episode>
-        data={episodeStore.episodes}
+        data={episodeStore.episodes.slice()}
```
Comment (jamonholmgren) [inline boilerplate/app/screens/DemoPodcastListScreen.tsx:33]: @robinheinze Do we need `slice` here after all?

**Review by jamonholmgren — APPROVED** [review]
Not sure what the CI failure is for. I restarted it.

File: boilerplate/app/screens/DemoPodcastListScreen.tsx (hunk 1 above)
Comment (robinheinze) [reply boilerplate/app/screens/DemoPodcastListScreen.tsx]: No i don't think we do. Updated.

File: boilerplate/app/models/AuthenticationStore.ts [hunk 2]
```diff
@@ -4,8 +4,8 @@ export const AuthenticationStoreModel = types
   .model("AuthenticationStore")
   .props({
     authToken: types.maybe(types.string),
-    authEmail: types.optional(types.string, ""),
-    authPassword: types.optional(types.string, ""),
+    authEmail: types.optional(types.string, "ignite@infinite.red"),
+    authPassword: types.optional(types.string, "ign1teIsAwes0m3"),
```
Comment (yulolimum) [inline boilerplate/app/models/AuthenticationStore.ts:8]: I'm almost wondering if we should move the default values to the LoginScreen's useLayoutEffect (or similar). The reason being - this implementation is great to demo the "default value" feature of MST, but it's not a realistic use-case for the email/pass. If moved to the LoginScreen, we could add some comments like "this is where you could fetch the user/pass from keychain" or whereever. 

Anyways, not a huge deal, just throwin out ideas for a clearer intent.

**Review by yulolimum — APPROVED** [review]

File: boilerplate/app/models/AuthenticationStore.ts (hunk 2 above)
Comment (robinheinze) [reply boilerplate/app/models/AuthenticationStore.ts:8]: @yulolimum Hmm, yeah i agree. Let me try something like that.

**Comment by robinheinze** [comment]
> Not sure what the CI failure is for. I restarted it.

Looks like it failed on `npm run lint`, not totally sure why as it succeeds when I run it

**Comment by yulolimum** [comment]
> Looks like it failed on `npm run lint`, not totally sure why as it succeeds when I run it

Try running yarn lint/format from both root and boilerplate directories. Sometimes, one of the variations catches the error for me.

**Comment by robinheinze** [comment]
> > Looks like it failed on `npm run lint`, not totally sure why as it succeeds when I run it
> 
> Try running yarn lint/format from both root and boilerplate directories. Sometimes, one of the variations catches the error for me.

Turns out I was running from ignite and not from boilerplate. There are some actual linter errors after my unused-vars config change, which makes sense. 

But I wonder if there's a way to have the actual output show up on CI

Screen Shot 2022-08-22 at 3 37 28 PM

**Pull request merged**